backend/src/main/java/com/map/service/GeocodingService.java

```diff
@@ -0,0 +1,24 @@
+package com.map.service;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Service
+public class GeocodingService {
+
+    @Value("${geocoding.api.key}")
+    private String apiKey;
+
+    public String buildUri(String address) {
```

Comment:
I'm not sure, but I think all the config of external api should go to app.properties, don't hardcode it here